plugin/src/main/kotlin/org/jetbrains/research/pynose/plugin/inspections/ObscureInLineSetupTestSmellInspection.kt

```diff
@@ -0,0 +1,70 @@
+package org.jetbrains.research.pynose.plugin.inspections
+
+import com.intellij.codeInspection.LocalInspectionToolSession
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import com.jetbrains.python.inspections.PyInspection
+import com.jetbrains.python.inspections.PyInspectionVisitor
+import com.jetbrains.python.psi.PyAssignmentStatement
+import com.jetbrains.python.psi.PyClass
+import com.jetbrains.python.psi.PyElementVisitor
+import com.jetbrains.python.psi.PyFunction
+import org.jetbrains.annotations.NotNull
+import org.jetbrains.research.pynose.plugin.util.TestSmellBundle
+import org.jetbrains.research.pynose.plugin.util.UnittestInspectionsUtils
+
+class ObscureInLineSetupTestSmellInspection : PyInspection() {
+    private val LOG = Logger.getInstance(ObscureInLineSetupTestSmellInspection::class.java)
+    private val testMethodLocalVarCount: MutableMap<PyFunction, MutableSet<String?>> = mutableMapOf()
+
+    override fun buildVisitor(
+        holder: ProblemsHolder,
+        isOnTheFly: Boolean,
+        @NotNull session: LocalInspectionToolSession
+    ): PyElementVisitor {
+
+        fun registerObscureInLineSetup(valueParam: PsiElement) {
+            holder.registerProblem(
+                valueParam,
+                TestSmellBundle.message("inspections.obscure.setup.description"),
+                ProblemHighlightType.WARNING
+            )
+        }
+
+        return object : PyInspectionVisitor(holder, session) {
+            override fun visitPyClass(node: PyClass) {
+                super.visitPyClass(node)
+                if (UnittestInspectionsUtils.isValidUnittestCase(node)) {
+                    UnittestInspectionsUtils.gatherUnittestTestMethods(node)
+                        .forEach { testMethod ->
+                            visitPyElement(testMethod)
+                        }
+                    testMethodLocalVarCount.keys
+                        .filter { x -> testMethodLocalVarCount[x]!!.size > 10 }
+                        .forEach { x ->
+                            registerObscureInLineSetup(x.nameIdentifier!!)
+                        }
+                    testMethodLocalVarCount.clear()
+                }
+            }
+
+            override fun visitPyAssignmentStatement(assignmentStatement: PyAssignmentStatement) {
+                super.visitPyAssignmentStatement(assignmentStatement)
+                val testMethod = PsiTreeUtil.getParentOfType(assignmentStatement, PyFunction::class.java)
+                if (!UnittestInspectionsUtils.isValidUnittestMethod(testMethod)) {
```

Comment:
Again, it's too expensive. You can get rid of it by processing test methods one by one in `visitPyClass`, without calling any other visitor's methods